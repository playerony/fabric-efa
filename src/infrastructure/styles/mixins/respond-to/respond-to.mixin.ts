import { css } from 'styled-components';

import { deviceBreakpoint } from '../..';
import { MediaFor, CSSParams } from './respond-to.type';

const keys = Object.keys(deviceBreakpoint) as Array<keyof typeof deviceBreakpoint>;

const respondToDecorator = (mediaFor: MediaFor) =>
  keys.reduce((accumulator, label) => {
    accumulator[label] = (...args: CSSParams) => css`
      @media (${mediaFor}: ${deviceBreakpoint[label]}px) {
        ${css(...args)};
      }
    `;

    return accumulator;
  }, {} as Record<keyof typeof deviceBreakpoint, Function>);

export const respondToMin = respondToDecorator('min-width');

export const respondToMax = respondToDecorator('max-width');
