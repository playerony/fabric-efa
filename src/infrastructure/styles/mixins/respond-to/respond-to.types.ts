import { css } from 'styled-components';

export type CSSParams = Parameters<typeof css>;
export type MediaFor = 'min-width' | 'max-width';
