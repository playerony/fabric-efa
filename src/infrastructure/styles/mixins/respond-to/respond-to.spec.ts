import { respondToMin, respondToMax } from './respond-to.mixin';

import { toMatchSnapshot, functionImportTest } from '@utils';

describe('respondToMin mixin', () => {
  functionImportTest(respondToMin.large);
  toMatchSnapshot(() => respondToMin.large`background-color: red;`);
});

describe('respondToMax mixin', () => {
  functionImportTest(respondToMax.large);
  toMatchSnapshot(() => respondToMax.large`background-color: red;`);
});
