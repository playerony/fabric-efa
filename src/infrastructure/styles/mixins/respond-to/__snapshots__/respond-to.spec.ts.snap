// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`respondToMax mixin snapshot test should match snapshot 1`] = `
Array [
  "
      @media (",
  "max-width",
  ": ",
  "1919",
  "px) {
        ",
  "background-color: red;",
  ";
      }
    ",
]
`;

exports[`respondToMin mixin snapshot test should match snapshot 1`] = `
Array [
  "
      @media (",
  "min-width",
  ": ",
  "1919",
  "px) {
        ",
  "background-color: red;",
  ";
      }
    ",
]
`;
